### DISCUSSION TITLE
chamilo y php8

### DISCUSSION THREAD

**Discussion by chatonito** [chamilo/chamilo-lms#5523] [General]
mi proveedor de hosting: hostgator México, no tiene soporte para php7.4, SOLO para php8.X y es un problema en la administración de chamilo que menciono a continuación:
importación de SCORM -->error 500
edición SCORM --> desfase del editor y componentes
Red social --> no se puede cambiar imagenes de los grupos error 500
plugin buycourses --> error 500 al querer consultar catálogo de cursos y membresías
importar usuarios --> NO FUNCIONA en php8 error 500

¿pueden ayudarnos a como resolver estas acciones?, son críticas para los profesores, la importación de lecciones y la edición de las mismas, especialmente porque no podemos bajar a php7.4 ya que el proveedor de hosting lo tiene descontinuado y tenemos que hacer el trabajo con php8.3 que nos da esos errores en chamilo

muchas gracias
atte
dr. agustin rico guzman
webmaster Enseñanza Médica CAR

**Comment by chatonito** [comment]
version de chamilo 1.11.26  php8.3 servidor Linux apache y mysql

**Comment by nosolored** [comment]
Chamilo 1.11.x no es del todo compatible con PHP 8, se ha comentado en anteriores ocasiones en el foro. No existe una solución fácil que implique cambiar una línea o algo así.

**Comment by Kaneda-1** [comment]
Otras soluciones:

- Contratar un hosting que te permita seleccionar desde el panel PHP 7.4.x y no te impoga la última versión 8.3.
- Un VPS o un servidor dedicado donde puedas utilizar la versión PHP 7.4.x.
- Financiar a desarrolladores para que Chamilo 1.x funcione con PHP 8.x (y luego liberar el código en github para beneficio de todos).
- Lanzar un crowdfunding (financiación colectiva) para proporcionar fondos suficientes a los desarrolladores para que Chamilo 1.x tenga compatibilidad con PHP 8.x.

Un saludo.

**Comment by wolmerhippe** [comment]
Try to insert these codes into the .htaccess file:

'# Set the “ea-php74” package as the default “PHP” programming language.
<'IfModule mime_module>
  AddHandler application/x-httpd-ea-php74___lsphp .php .php7 .phtml
<'/IfModule>

It worked for me.

**Reply by Kaneda-1** [reply]
Nota:

```
php – BEGIN cPanel-generated handler, do not edit
Set the “ea-php74” package as the default “PHP” programming language.
<IfModule mime_module>
AddHandler application/x-httpd-ea-php74___lsphp .php .php7 .phtml
</IfModule>
php – END cPanel-generated handler, do not edit
```

EA-PHP son las versiones de PHP que cPanel empaqueta y proporciona a través de los repositorios EasyApache 4. 
Estas son las versiones estándar que se incluyen en una instalación base de cPanel y se administran a través del Administrador MultiPHP en la interfaz de cPanel y WHM.

EA-PHP (ea-php74) is the PHP versions that cPanel packages and provides via the EasyApache 4 repositories. 
These are the standard versions that are included in a base installation of cPanel and are managed via the MultiPHP Manager in both cPanel & WHM interface.

**Reply by wolmerhippe** [reply]
Yes. But some providers (like mine) omit this option (7.4). And it accepts the change into the .htaccess file.
